x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/use_rule_details_flyout.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback } from 'react';
+import { invariant } from '../../../../../common/utils/invariant';
+import type {
+  RuleInstallationInfoForReview,
+  RuleSignatureId,
+} from '../../../../../common/api/detection_engine';
+import type { DiffableRule } from '../../../../../common/api/detection_engine/prebuilt_rules/model/diff/diffable_rule/diffable_rule';
+
+export interface RuleDetailsFlyoutState {
+  flyoutRule: RuleInstallationInfoForReview | null;
+}
+
+export interface RuleDetailsFlyoutActions {
+  openFlyoutForRuleId: (ruleId: RuleSignatureId) => void;
+  closeFlyout: () => void;
+}
+
+export const useRuleDetailsFlyout = (
+  rules: DiffableRule[]
+): RuleDetailsFlyoutState & RuleDetailsFlyoutActions => {
+  const [flyoutRule, setFlyoutRule] = React.useState<RuleInstallationInfoForReview | null>(null);
+
+  const openFlyoutForRuleId = useCallback(
+    (ruleId: RuleSignatureId) => {
+      const ruleToShowInFlyout = rules.find((rule) => rule.rule_id === ruleId);
```

Comment:
This is the only way to find rules that are not installed yet - they only have this `RuleSignatureId`. For the rule upgrade table, I think a better way would be finding rules by their `RuleObjectId`. Signature ids sometimes can be duplicated in multiple rules.

Also I think a hook for the upgrade table shouldn't depend on the `RuleInstallationInfoForReview` type. There's a `RuleUpgradeInfoForReview` type which is used in `UpgradePrebuiltRulesTableContextProvider`.